pkg/operator/apis/config/v1alpha1/defaults_test.go

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) 2022 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1_test
+
+import (
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	. "github.com/onsi/gomega/gstruct"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	componentbaseconfigv1alpha1 "k8s.io/component-base/config/v1alpha1"
+	"k8s.io/utils/pointer"
+
+	"github.com/gardener/gardener/pkg/logger"
+	. "github.com/gardener/gardener/pkg/operator/apis/config/v1alpha1"
+)
+
+var _ = Describe("Defaults", func() {
+	Describe("OperatorConfiguration", func() {
+		var obj *OperatorConfiguration
+
+		BeforeEach(func() {
+			obj = &OperatorConfiguration{}
+		})
+
+		It("should correctly default the configuration", func() {
+			SetObjectDefaults_OperatorConfiguration(obj)
+
+			Expect(obj.LogLevel).To(Equal(logger.InfoLevel))
+			Expect(obj.LogFormat).To(Equal(logger.FormatJSON))
+
+			Expect(obj.Server.HealthProbes.BindAddress).To(BeEmpty())
+			Expect(obj.Server.HealthProbes.Port).To(Equal(2750))
+			Expect(obj.Server.Metrics.BindAddress).To(BeEmpty())
+			Expect(obj.Server.Metrics.Port).To(Equal(2751))
+		})
+
```

Comment:
Can you add another test that verifies any custom settings like you did for the leader election below?